src/ctia/schemas/search_agg.clj

```diff
@@ -0,0 +1,62 @@
+(ns ctia.schemas.search-agg
+  (:require [schema.core :as s]
+            [schema-tools.core :as st]))
+
+(s/defschema DateRange
+  {s/Keyword
+   (st/optional-keys
+    {:gte s/Inst
+     :lt s/Inst})})
+
+(s/defschema SearchQuery
+  (st/optional-keys
+   {:query-string s/Str
+    :filter-map {s/Keyword s/Any}
+    :date-range DateRange}))
+
+(s/defschema AggType (s/enum :histogram :topn :cardinality))
+
+(s/defschema AggCommonParams
+  (st/merge
+   {:aggregate-on s/Str}))
+
+(s/defschema Timezone
+  (let [positives (map #(format "+%02d:00" %) (range 12))
+        negatives (map #(format "-%02d:00" %) (range 1 12))]
+    (->> (concat positives negatives)
+         (apply s/enum))))
+
+(s/defschema HistogramParams
+  (st/merge
+   AggCommonParams
+   {:granularity (s/enum :day :week :month)
+    (s/optional-key :timezone) Timezone}))
+
+(s/defschema HistogramQuery
```

Comment:
Maybe some of these defschemas could have a docstring. The names (`HistogramQuery`) seem pretty self-explanatory tbh, although sometimes some extra context can be added